engine/signalmgr/queries.sql

```diff
@@ -0,0 +1,34 @@
+-- name: SignalMgrGetPending :many
+-- Get a batch of pending signals to process.
+SELECT
+    id,
+    dest_id,
+    service_id
+FROM
+    pending_signals
+WHERE
+    message_id IS NULL
+FOR UPDATE
+    SKIP LOCKED
+LIMIT 100;
+
+-- name: SignalMgrInsertMessage :exec
+-- Insert a new message into the outgoing_messages table.
+INSERT INTO outgoing_messages(id, message_type, service_id, channel_id)
+    VALUES ($1, 'signal_message', $2, $3);
+
+-- name: SignalMgrUpdateSignal :exec
+-- Update a pending signal with the message_id.
+UPDATE
+    pending_signals
+SET
+    message_id = $2
+WHERE
+    id = $1;
+
+-- name: SignalMgrDeleteStale :exec
+-- Delete stale pending signals.
+DELETE FROM pending_signals
+WHERE message_id IS NULL
+    AND created_at < NOW() - INTERVAL '1 hour';
```

Comment:
I feel like this may have unexpected consequences down the road, but hopefully a user is never in this situation

Reply:
Mitigating factors:
- The data in this table is only useful/needed until a message is sent
- If a message is over an hour old, it means there is at least an hour worth of more-important messages being sent in the meantime
- Signals represent a way to potentially have a lot more messages coming through, so it serve as a cap on the backlog as well